scio-examples/src/main/scala/com/spotify/scio/examples/extra/RedisExamples.scala

```diff
@@ -57,6 +60,43 @@ object RedisReadStringsExample {
 
 }
 
+// ## Redis Lookup Strings example
+
+// Usage:
+
+// `sbt "runMain com.spotify.scio.examples.extra.RedisLookUpStringsExample
+// --project=[PROJECT] --runner=DataflowRunner --zone=[ZONE]
+// --redisHost=[REDIS_HOST]
+// --redisPort=[REDIS_PORT]
+object RedisLookUpStringsExample {
+
+  private def lookupFn(connectionOptions: RedisConnectionOptions) =
+    new RedisLookupDoFn[String, Option[String]](connectionOptions) {
+      override def asyncLookup(
+        client: ThreadLocal[Jedis],
```

Comment:
Hi @stormy-ua what about using `JedisPool`? It seems that it's intended for multi-threaded envs.

Also stumbled upon this https://github.com/lettuce-io/lettuce-core the other day and it seems that it would be a better fit here since it supports async calls.

We can start with `Jedis` and at some point switch to `lettuce`

Reply:
I can look into the `JedisPool`. The issue with the `JedisPool` is that a `JedisClient` must be explicitly returned to the pool after it is no longer needed. It isn't possible to implement this with the `BaseAsyncLookupDoFn`. I can look into how the `BaseAsyncLookupDoFn` could be changed or extended to support an explicit resource cleanup handling. 

As for `lettuce`, it looks very interesting and I think this might be a better way to go forward. I sticked to `jedis`, because this is what the original Beam Redis IO used, but we can easily replace it.

Reply:
Hi @stormy-ua I was thinking about this a little bit and I think we can have an alternative take on this. We don't really need to make this an `async` lookup... we can still leverage `Jedis` and it's `Pipeline` for batch requests. The idea here is to generalize a little bit more `RedisWriteFn` into a `RedisFn` that we can use for lookups as well. I'll put up smth soon so we can see how it would look like and discuss.

Reply:
@regadas Sounds good! This would be the best option if we can just use `Jedis`/`Pipeline`.